src/gui/tray/ActivityListModel.cpp

```diff
@@ -444,13 +445,19 @@ void ActivityListModel::triggerDefaultAction(int activityIndex) const
 
         const auto baseName = QFileInfo(basePath).fileName();
 
-        ConflictDialog dialog;
-        dialog.setBaseFilename(baseName);
-        dialog.setLocalVersionFilename(conflictedPath);
-        dialog.setRemoteVersionFilename(basePath);
-        if (dialog.exec() == ConflictDialog::Accepted) {
-            folder->scheduleThisFolderSoon();
+        if (!_currentConflictDialog.isNull()) {
+            _currentConflictDialog->close();
         }
+        _currentConflictDialog = new ConflictDialog;
+        _currentConflictDialog->setBaseFilename(baseName);
+        _currentConflictDialog->setLocalVersionFilename(conflictedPath);
+        _currentConflictDialog->setRemoteVersionFilename(basePath);
+        _currentConflictDialog->setAttribute(Qt::WA_DeleteOnClose);
+        connect(_currentConflictDialog, &ConflictDialog::accepted, [folder]() {
```

Comment:
For extra safety, `folder` should be passed as third parameter of that connect just before the lambda. In case the folder object would be deleted before accepted is emitted it would thus disconnect automatically, otherwise it won't and you risk calling into a deleted object from the lambda.